visualize-admin/visualization-tool
# feat: Dashboards improvements

This PR introduces optimizations and fixes related to dashboards:
- bigger charts in published mode,
- responsive tall layout (switching to one column on mobile devices and aligning chart elements along columns in a row),
- filters moved below the chart,
- fixed table previews and interactive filters in preview mode.

### 🟰 Alignment of chart elements across columns
Generally, charts that are next to each other in a Tall layout should:
- respect maximum common height of the "header" (title + description + interactive filters button) so that the chart SVGs are aligned,
- respect maximum common chart SVG height, so that the chart footnotes are aligned.

For visual reference see [Figma design](https://www.figma.com/file/Hn9wvuEYLUmdJtg4KwZYUx/visualize.admin.ch---design?type=design&node-id=2586-154739&mode=design&t=8tE6QCc5u7o9MmSx-4).

### How to test
1. Go to [this link](https://visualization-tool-git-feat-improve-responsiveness-ixt1.vercel.app/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/9&dataSource=Prod&flag__layoutStep=true).
2. Add two more charts, so there are three in total.
3. Try to add long titles to two charts.
4. Enable an interactive filter in one chart.
5. Proceed to layout options.
6. Change layout to Dashboard -> Tall.
7. Switch to table preview for every of the chart and see that it works correctly (while on TEST the table layout is applied for every chart and breaks their sizes).
8. Go back to chart modes.
9. See that the interactive filter works.
10. See that the chart elements are aligned (for two charts next to each other).
11. Resize the window to see that the charts stack on top of each other when window width is small.
